mineral-ui/mineral-ui
# Component display names

### Description

Addresses a couple of issues with regards to component displayNames

* Stop setting `displayName` on `element` passed to `createStyledComponent`.  It is a static property and this was incorrectly overwriting values when the `element` was a component. This should have been set on the returned component rather than the element. As the call to Emotion/styled already does this for us, we actually don’t need to do anything at all.
* Manually add component displayNames to library components.  This was also implemented using babel plugin but we decided against it as it had to be done in the website too (> bundle size) and also the plugin did not have babel 7 compatibility.

### Motivation and context

* Component displayNames were being erroneously set in `createStyledComponent` and it could cause confusion, particularly when debugging.
* Adding displayNames to components allows us to use those values in production.  Mineral uses them in several places for loose/flexible type checking, such as in `ButtonGroup` and `FormField`.  

### Screenshots, videos, or demo, if appropriate

https://fix-display-names--mineral-ui.netlify.com/

### How to test

1. No change in functionality
2. Validate that ButtonGroup still works appropriately.  It uses a displayName type check.
4. View the snapshots and see that they are inline with expectations
5. Have a look at dev-tools and see that component names look correct, and also that dev mode CSS classNames are still labelled correctly.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist
